tests/system/providers/google/cloud/dataflow/example_dataflow_go_deferrable.py

```diff
@@ -0,0 +1,156 @@
+# under the License.
+"""
+Example Airflow DAG for Apache Beam operators
+
+Requirements:
+    This test requires the gcloud and go commands to run.
+"""
+from __future__ import annotations
+
+import os
+from datetime import datetime
+from pathlib import Path
+
+from airflow.models.dag import DAG
+from airflow.providers.apache.beam.hooks.beam import BeamRunnerType
+from airflow.providers.apache.beam.operators.beam import BeamRunGoPipelineOperator
+from airflow.providers.google.cloud.hooks.dataflow import DataflowJobStatus
+from airflow.providers.google.cloud.operators.dataflow import DataflowConfiguration
+from airflow.providers.google.cloud.operators.gcs import GCSCreateBucketOperator, GCSDeleteBucketOperator
+from airflow.providers.google.cloud.sensors.dataflow import (
+    DataflowJobAutoScalingEventsSensor,
+    DataflowJobMessagesSensor,
+    DataflowJobStatusSensor,
+)
+from airflow.providers.google.cloud.transfers.local_to_gcs import LocalFilesystemToGCSOperator
+from airflow.utils.trigger_rule import TriggerRule
+
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
+DAG_ID = "dataflow_native_go_async"
+BUCKET_NAME = f"bucket_{DAG_ID}_{ENV_ID}"
+
+GCS_TMP = f"gs://{BUCKET_NAME}/temp/"
+GCS_STAGING = f"gs://{BUCKET_NAME}/staging/"
+GCS_OUTPUT = f"gs://{BUCKET_NAME}/output"
+
+GO_FILE_NAME = "wordcount.go"
+GO_FILE_LOCAL_PATH = str(Path(__file__).parent / "resources" / GO_FILE_NAME)
+GCS_GO = f"gs://{BUCKET_NAME}/{GO_FILE_NAME}"
+LOCATION = "europe-west3"
+
+default_args = {
+    "dataflow_default_options": {
+        "tempLocation": GCS_TMP,
+        "stagingLocation": GCS_STAGING,
+    }
+}
+
+with DAG(
+    "example_beam_native_go_deferrable",
+    start_date=datetime(2021, 1, 1),
+    schedule="@once",
+    catchup=False,
+    default_args=default_args,
+    tags=["example"],
+) as dag:
+    create_bucket = GCSCreateBucketOperator(task_id="create_bucket", bucket_name=BUCKET_NAME)
+
+    upload_file = LocalFilesystemToGCSOperator(
+        task_id="upload_file_to_bucket",
+        src=GO_FILE_LOCAL_PATH,
+        dst=GO_FILE_NAME,
+        bucket=BUCKET_NAME,
+    )
+
+    start_go_pipeline_dataflow_runner = BeamRunGoPipelineOperator(
+        task_id="start_go_pipeline_dataflow_runner",
+        runner=BeamRunnerType.DataflowRunner,
+        go_file=GCS_GO,
+        pipeline_options={
+            "tempLocation": GCS_TMP,
+            "stagingLocation": GCS_STAGING,
+            "output": GCS_OUTPUT,
+            "WorkerHarnessContainerImage": "apache/beam_go_sdk:2.46.0",
+        },
+        dataflow_config=DataflowConfiguration(job_name="start_go_job", location=LOCATION),
+    )
+
+    # wait_for_go_job_async_done = DataflowJobStatusSensor(
+    #     task_id="wait_for_go_job_async_done",
+    #     job_id="{{task_instance.xcom_pull('start_go_pipeline_dataflow_runner')['dataflow_job_id']}}",
+    #     expected_statuses={DataflowJobStatus.JOB_STATE_DONE},
+    #     location=LOCATION,
+    # )
```

Comment:
Do we need this piece of code?

Reply:
This is one of the questions that I listed in my PR description. It is not clear to me what to do with the example dags while 2 Dataflow operators are still not implemented. 
If we decide to not include it into the final PR - then I will remove this.

Reply:
I think it will be better to include this `example_dags` when we have an implementation for these 2 Dataflow operators.

Reply:
I agree. File removed.